### PR TITLE
feat(gpu): automatic detection of optimal GPU params

### PR DESCRIPTION
**Pull request by keyvank**
This PR tries to implement auto-detection of best GPU parameters for multiexp algorithm based on the card name and input size.

### CODE REVIEW

**Comment by dignifiedquire** [comment]
got some conflicts after I merged the other PR

File: src/gpu/utils.rs [hunk 1]
```diff
@@ -21,3 +21,22 @@ pub fn get_devices(platform_name: &str) -> GPUResult<Vec<Device>> {
         None => Err(GPUError {msg: "GPU platform not found!".to_string() })
     }
 }
+
+use std::collections::HashMap;
+lazy_static!{
+    static ref CORE_COUNTS: HashMap<&'static str, usize> = vec![
+        ("GeForce RTX 2080 Ti", 4352),
```
Comment (dignifiedquire) [inline src/gpu/utils.rs]: can't the cards report their own core count Oo

Comment (keyvank) [reply src/gpu/utils.rs]: I couldn't get the right numbers for this from clinfo unfortunately.

Comment (keyvank) [reply src/gpu/utils.rs]: but this also prevents running the kernel on unknown, untested cards which is good i think  :smiley:

**Pull request merged**